src/Knp/Menu/Matcher/Voter/RouteVoter.php

```diff
@@ -91,4 +99,25 @@ private function isMatchingRoute(Request $request, array $testedRoute): bool
 
         return true;
     }
+
+    /**
+     * @phpstan-param array{route?: string|null, pattern?: string|null, parameters?: array<string, mixed>, query_parameters?: array<string, mixed>} $testedRoute
+     */
+    private function isMatchingQueryParameters(Request $request, array $testedRoute): bool
+    {
+        if (!isset($testedRoute['query_parameters'])) {
+            return true;
+        }
+
+        $routeQueryParameters = $request->query->all();
+
+        foreach ($testedRoute['query_parameters'] as $name => $value) {
+            // cast both to string so that we handle integer and other non-string parameters, but don't stumble on 0 == 'abc'.
+            if (!isset($routeQueryParameters[$name]) || (string) $routeQueryParameters[$name] !== (string) $value) {
```

Comment:
be careful. `$routeQueryParameters[$name]` could be an array if the request passes it multiple times. We should handle that carefully rather than casting the array to string (to me, we should make it return false)